Extensions/TileMap/collision/TransformedTileMap.ts

```diff
@@ -733,30 +734,130 @@ namespace gdjs {
         }
         tileTransformation.preConcatenate(layerTransformation);
 
-        // The tile map can't change at runtime so the existing arrays can be
-        // reused safely.
+        const hasFullHitBox =
+          definitionHitboxes.length === 1 && definition.hasFullHitBox(tag);
+        if (hasFullHitBox) {
```

Comment:
Typically, that's something I would like moved in a function which will, by its name itself, document what's going on (you don't need to read the full 75 new lines before you understand "what the heck is going on", the name of the function will help you, and this can be tested or refactored later)

Reply:
I didn't split the function but I added comments (and sort the code a bit) because I think it can be a critical path and I want to avoid accessing 3 times to the same context.
Also, I think that tests would check the hit boxes returned by the object. I don't see value in checking intermediary steps that are implementation details.